pygmt/tests/test_grd2cpt.py

```diff
@@ -0,0 +1,344 @@
+    fig_test.colorbar(frame="a2000")
+    return fig_ref, fig_test
+
+
+@check_figures_equal()
+def test_grd2cpt_continuous(grid):
+    """
+    Create a CPT with that is continuous.
+    """
+    fig_ref, fig_test = Figure(), Figure()
+    # Use single-character arguments for the reference image
+    fig_ref.basemap(B="a", J="W0/15c", R="d")
+    grd2cpt(grid=grid, C="blue,white", Z=True)
+    fig_ref.colorbar(B="a2000")
+    fig_test.basemap(frame="a", projection="W0/15c", region="d")
+    grd2cpt(grid=grid, cmap="blue,white", continuous=True)
+    fig_test.colorbar(frame="a2000")
+    return fig_ref, fig_test
+
+
+@check_figures_equal()
+def test_grd2cpt_categorical(grid):
+    """
+    Create a CPT with that is categorical.
+    """
+    fig_ref, fig_test = Figure(), Figure()
+    # Use single-character arguments for the reference image
+    fig_ref.basemap(B="a", J="W0/15c", R="d")
+    grd2cpt(grid=grid, C="geo", W=True)
+    fig_ref.colorbar(B="a2000")
+    fig_test.basemap(frame="a", projection="W0/15c", region="d")
+    grd2cpt(grid=grid, cmap="geo", categorical=True)
+    fig_test.colorbar(frame="a2000")
+    return fig_ref, fig_test
+
+
+@check_figures_equal()
+def test_grd2cpt_cyclic(grid):
+    """
+    Create a CPT with that is cyclic.
+    """
+    fig_ref, fig_test = Figure(), Figure()
+    # Use single-character arguments for the reference image
+    fig_ref.basemap(B="a", J="W0/15c", R="d")
+    grd2cpt(grid=grid, C="geo", Ww=True)
+    fig_ref.colorbar(B="a2000")
+    fig_test.basemap(frame="a", projection="W0/15c", region="d")
+    grd2cpt(grid=grid, cmap="geo", cyclic=True)
+    fig_test.colorbar(frame="a2000")
+    return fig_ref, fig_test
+
+
+@check_figures_equal()
+def test_grd2cpt_limit(grid):
+    """
+    Create a CPT with that sets a min/max limit.
+    """
+    fig_ref, fig_test = Figure(), Figure()
+    # Use single-character arguments for the reference image
+    fig_ref.basemap(B="a", J="W0/15c", R="d")
+    grd2cpt(grid=grid, C="haxby", L="-1000/1000")
+    fig_ref.colorbar(B="a2000")
+    fig_test.basemap(frame="a", projection="W0/15c", region="d")
+    grd2cpt(grid=grid, cmap="haxby", limit=[-1000, 1000])
+    fig_test.colorbar(frame="a2000")
+    return fig_ref, fig_test
+
+
+def test_grd2cpt_blank_output(grid):
+    """
+    Use incorrect setting by passing in blank file name to output parameter.
+    """
+    with pytest.raises(GMTInvalidInput):
+        grd2cpt(grid=grid, output="")
+
+
+def test_grd2cpt_invalid_output(grid):
+    """
+    Use incorrect setting by passing in invalid type to output parameter.
+    """
+    with pytest.raises(GMTInvalidInput):
+        grd2cpt(grid=grid, output=["some.cpt"])
+
+
+def test_grd2cpt_output_to_cpt_file(grid):
+    """
+    Save the generated static color palette table to a .cpt file.
+    """
+    with GMTTempFile(suffix=".cpt") as cptfile:
+        grd2cpt(grid=grid, output=cptfile.name)
+        assert os.path.exists(cptfile.name)
```

Comment:
This test doesn't work, because when calling `with GMTTempFile(suffix=".cpt")`, an empty temporary file is always created, no matter you call `grd2cpt` or not.

Reply:
Removed in 72130a9

Reply:
@willschlitzer Sorry if my previous comment is misleading. The test is still necessary. The point is we can't check if the CPT is written using `assert os.path.exists(cptfile.name)`, because the temporary file `cptfile.name` always exists but with a zero size. I think at least we can check if the file size is non-zero.

Reply:
I changed it to `assert os.path.getsize(cptfile.name) > 0` and that is passing! Just to confirm is wasn't a fluke I tested `assert os.path.getsize(cptfile.name) == 0` and the test failed.

Reply:
> I changed it to `assert os.path.getsize(cptfile.name) > 0` and that is passing!

It looks good to me. Ideally we should check the content of the CPT file, but it needs more coding, and I think we can leave it to future.